sound/soc/dwc/dwc-i2s.c

```diff
@@ -210,10 +212,6 @@ static void dw_i2s_config(struct dw_i2s_dev *dev, int stream)
 			dmacr |= (DMACR_DMAEN_RXCH0 << ch_reg);
```

Comment:
It looks to me with the old code, unused channels would be cleared (by the `dmacr=0` initialisation and then not setting the bit). Does this still happen with new code? (e.g. if this is called once with `config->chan_nr=4` and then with `config->chan_nr=2`.

(Note: I'm not familiar with how this code is called, so ignore if you are happy with new behaviour).

Reply:
That's annoyingly correct. I've added clearing of the relevant group of channel enable flags before setting the appropriate number.